frontend/src/components/MissionOverview/UpcomingMissionCard.tsx

```diff
@@ -26,13 +27,23 @@ const StyledMissionStartTime = styled.div`
 Icon.add({ more_vertical })
 
 export function UpcomingMissionCard({ mission }: MissionProps) {
+    let navigate = useNavigate()
+    const routeChange = () => {
+        let path = '/mission/' + mission.id
+        navigate(path)
+    }
     return (
-        <StyledMissionCard key={mission.id} variant="default" style={{ boxShadow: tokens.elevation.raised }}>
+        <StyledMissionCard
+            key={mission.id}
+            variant="default"
+            style={{ boxShadow: tokens.elevation.raised }}
+            onClick={routeChange}
+        >
             <HorizontalContent>
                 <Checkbox />
                 <Typography variant="h6">Mission name</Typography>
                 <MissionStartTimeDisplay mission={mission} />
-                <Typography>Tags: 17</Typography>
+                <Typography>Tags: {mission.plannedTasks.length}</Typography>
```

Comment:
Also, a bit confusing to say "Tags: 1" and then the mission containts 2 tasks (because there are two inspections on the same tag)

Reply:
Changed to "Tasks: 2"